### DISCUSSION TITLE
Enhancement: New features on the Anthropic API reducing costs by up to 90%.

### DISCUSSION THREAD

**Discussion by tracer8** [danny-avila/LibreChat#5034] [Feature Requests & Suggestions]
### What features would you like to see added?

New features on the Anthropic API reducing costs by up to 90%.

> Several features are now generally available on the Anthropic API to help you better optimize for cost and speed, including: 

 

[Prompt caching](https://links.iterable.com/u/click?_t=67101f7d500248b386768eecaae4611b&_m=93cd147fbe9346edb44ec0bbea924e29&_e=MmOlgqRUYQ5j0Ypv4fK30lNkFnSU0Gw6tx3YFxk9jbVn1WP2xFk6zkkk3fFDhq71KpAHwE3kS7gKLcMhMXLTjbcYeDcKjqkSSsK2wuYEzPAsHeaVZksEl-SVepD9mufx3ghNUGh63lap028O3Zb71PBj5DWn3InprNf_K_PZQ7_Ti6rlqPqHFUZStAifyau_DJY0YHB7wH5IuNxACUoRzWG_3BFnHy0nbAR4OVmdkdXQfdWSDdacpNeRu7VyqhC4o9qn_OVLzoYMFUgkFDy0vg%3D%3D) 

Provide Claude with more background knowledge and example outputs to improve response accuracy and latency—all while reducing costs by up to 90%.

 

[Message Batches API](https://links.iterable.com/u/click?_t=67101f7d500248b386768eecaae4611b&_m=93cd147fbe9346edb44ec0bbea924e29&_e=MmOlgqRUYQ5j0Ypv4fK30lNkFnSU0Gw6tx3YFxk9jbVn1WP2xFk6zkkk3fFDhq71Vi_PaAUcJsnMssYCg8GEU0uj4Q0J6MO636DFUCTq6UUnxXST3YKF767mHgme9-bRsf2-kV_2aaxpAIrNGc2G5CrOJHUJdZ08ZWugXJUpHUDX4j5BV5GXNk_EL9A7UeNJiL45XbdBuoTNgtXUkjuDU0Y05jaCLfFFiByQycd2EymxBue3dmJ5U1ac4JkLS6-_1NxusSBxaKXe02ovEqg-JulzR29WhKjrKdTg8YAG6uM%3D) 

Send any number of batches of up to 100,000 messages per batch. Batches are processed asynchronously with results returned as soon as the batch is complete and cost 50% less than standard API calls.

 

[Token counting](https://links.iterable.com/u/click?_t=67101f7d500248b386768eecaae4611b&_m=93cd147fbe9346edb44ec0bbea924e29&_e=MmOlgqRUYQ5j0Ypv4fK30lNkFnSU0Gw6tx3YFxk9jbVn1WP2xFk6zkkk3fFDhq71fuT4PsarR1lJAp9IMtd9iqRk00hp8QOunysInVsdTYl9y4JxgRrufGknOKPLWiVDWNexkDWtpVC7MSlHGTaPDDfv3pnXDctt0dQSf_clW5kQ2ccK3-Q1d2KuplY3fvxcAYkfQDG3YuBzVHkG9k_iwsgnVJ_ok6ynNTeO8H9uhnpFI7SNauv-QnhQQHY5xBSZq9BrvW4LCAXm1E69odNuMQ%3D%3D)

Determine the number of tokens in a message before sending it to Claude to make informed decisions about your prompts and usage.

 

[Multimodal PDF support](https://links.iterable.com/u/click?_t=67101f7d500248b386768eecaae4611b&_m=93cd147fbe9346edb44ec0bbea924e29&_e=MmOlgqRUYQ5j0Ypv4fK30lNkFnSU0Gw6tx3YFxk9jbVn1WP2xFk6zkkk3fFDhq71eQ1RTkOYF1MWg-xAq_FOgF_Tr2O5qopAqvmMfD5FwQat7VFbu3PLk6Iu6WpE-Ke46GqiF3YyFZ1Zb46F2id0Ox2l5Bkbkz8gk3_IZZ1pNFsaZ4qZI-_cGPkhfE4PYsquySbTTZqfH2Pt8c33TduwugjxcToOUBbj2aX5fyYra2m-RDhQi_plLhedr_cufLyX-qarrLlDap1YZE_dY786hQ%3D%3D) 

Understand and analyze both text and visual content within PDF documents for more comprehensive analysis. 

 

[Models endpoint](https://links.iterable.com/u/click?_t=67101f7d500248b386768eecaae4611b&_m=93cd147fbe9346edb44ec0bbea924e29&_e=MmOlgqRUYQ5j0Ypv4fK30lNkFnSU0Gw6tx3YFxk9jbU99tOHeBkOyd2GYsQUDmSa1CNiXOagqavUjES8I6XmiCkx06pzQOlrdQP982C8gZkV-y6OpQzbMtPUqetov-zfj-_EPbug0qnWuGnpTWMwWkw_LEDkyDLeHRP_Y7TSA5PMwcfqYX0EZ65Iv6bNxkm6Gx-wDm2B4ypmNfW_o7UkEDv8yyrCTEOo7xE45fMSD6KiNaIk8qIpWDHMXj-zVSxj)

List all available models to select between the latest Anthropic Models or [query a specific model](https://links.iterable.com/u/click?_t=67101f7d500248b386768eecaae4611b&_m=93cd147fbe9346edb44ec0bbea924e29&_e=MmOlgqRUYQ5j0Ypv4fK30lNkFnSU0Gw6tx3YFxk9jbU99tOHeBkOyd2GYsQUDmSaxnvgtb6G7cSrH50DN1SZvMijEI0lrxk6zHUr_aLmZk4QPmY3MFeQBV2TJ756KvEQUj7upT95OO71ushB5g4tUOS7LLI1cBJXIzlUX9r6nki5ukXO-rLA2POJiRQjp8zw4yJeCGc3Qcla5TA6j8EMolDUyvA71qVQlTH5qNKdyXOG8dC6msjLH-stZa12lcj0) string to validate user-provided content.

### More details

New features on the Anthropic API reducing costs by up to 90%.

### Pictures

![Screenshot_2024-12-18 11 16 31_001](https://github.com/user-attachments/assets/59c5a881-ada3-4fec-81a1-03997fc7693f)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Thanks. Prompt caching is already supported and some of these features already have open issues.

**Reply by tracer8** [reply]
Is caching automatic or i need to enable it by myself?

**Reply by danny-avila** [reply]
automatic and can be disabled

**Reply by tracer8** [reply]
thank you!
